svg-2-android-vector/src/main/java/com/quittle/svg2androidvector/Svg2AndroidVectorPlugin.java

```diff
@@ -53,14 +56,25 @@ public void apply(final Project project) {
     }
 
     private static void onAfterEvaluate(final Project project, final Svg2AndroidVectorExtension extension) {
+        // The base Android extension, where all resources can be found.
+        final BaseExtension androidExtension = project.getExtensions().findByType(BaseExtension.class);
+        if (androidExtension == null) {
+            // This might mean it was applied to a non-Android project, possibly just the root project.
+            return;
+        }
         final TaskContainer taskContainer = project.getTasks();
         final Task parentTask = taskContainer.create(CONVERSION_PARENT_TASK_NAME);
         // An early Android task all the conversion tasks should be a dependency of
-        taskContainer.findByName(EARLY_ANDROID_TASK_NAME).dependsOn(parentTask);
```

Comment:
Why does relying on `preBuild` no longer work? If you really don't need it anymore can you delete the constant as well?

Reply:
TBH, it probably can work, which is why I split that commit out. I'd never seen it done that way before (the "old" way was to do `variant.getPreBuild()`, the new way `variant.getPreBuildProvider().configure {}`), and I think it might trigger eager-evaluation of the task, but it probably works OK. Feel free to drop this patch if it is clearer without it (which I think it might be, even if it's somewhat less idiomatic)

Reply:
I returned to the original code here.